synapse/app/generic_worker.py

```diff
@@ -514,13 +515,13 @@ def _get_federation_out_pos(self, db_conn):
 class GenericWorkerServer(HomeServer):
     DATASTORE_CLASS = GenericWorkerSlavedStore
 
-    def _listen_http(self, listener_config):
-        port = listener_config["port"]
-        bind_addresses = listener_config["bind_addresses"]
-        site_tag = listener_config.get("tag", port)
+    def _listen_http(self, listener_config: ListenerConfig):
+        port = listener_config.port
+        bind_addresses = listener_config.bind_addresses
+        site_tag = listener_config.http_options.tag or port
```

Comment:
This is a slight change in behavior that might not matter (is `0` a valid tag?)

In `synapse.http.site` an assertion was added to ensure that `http_options` is not `None`. Curious why one was added there and not here?

Reply:
> This is a slight change in behavior that might not matter (is 0 a valid tag?)

it probably doesn't matter, but have updated it anyway.

> In synapse.http.site an assertion was added to ensure that http_options is not None. Curious why one was added there and not here?

I added one there to keep mypy happy. I'll add one here for consistency.